src/platform/android/DnssdImpl.cpp

```diff
@@ -242,8 +264,79 @@ void HandleResolve(jstring instanceName, jstring serviceType, jstring address, j
     CopyString(service.mName, jniInstanceName.c_str());
     CopyString(service.mType, jniServiceType.c_str());
     service.mPort = static_cast<uint16_t>(port);
+    service.mTextEntrySize = 0;
+    service.mTextEntries   = nullptr;
+
+    // Note on alloc/free memory use
+    // We are only allocating the entries list and the data field of each entry
+    // so we free these in the exit section
+    if (textEntries != nullptr)
+    {
+        jobjectArray keys   = (jobjectArray) env->CallObjectMethod(sMdnsCallbackObject, sGetTextEntryKeysMethod, textEntries);
+        size_t size         = env->GetArrayLength(keys);
+        TextEntry * entries = new (std::nothrow) TextEntry[size];
+        if (entries == nullptr)
+        {
+            ChipLogError(Discovery, "entries alloc failure");
+            goto exit;
+        }
+        service.mTextEntries = entries;
+        for (size_t i = 0; i < size; i++)
```

Comment:
please `memset(entries, 0, sizeof(entries[0])*size)` or use calloc.

in exit you do a cmp with nullptr so you don't want to free random memory.